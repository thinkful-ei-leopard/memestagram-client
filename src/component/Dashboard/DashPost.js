import React, { Component } from 'react'

export default class DashboardPage extends Component {
    state={
        count:0,
        comment:0,
        like:'🤍'
    }

    addLike(){
    
    if(this.state.like==='🤍'){
     this.setState({
         like:'❤️',
         count:this.state.count+1
     })
        
    }else if(this.state.like==='❤️'){
        this.setState({
            like:'🤍',
            count:this.state.count-1
        })
        
    }
    
   }
       
    
    render() {
        
        return (
            <div>
                <div className='profImg'>Image</div><span>Username</span>
                <div className='MemePost'>Mmem post</div>
        <div ><span role='img' aria-label='heart' onClick={() =>this.addLike()}>{this.state.like}</span>like: {this.state.count}<span> Comment: {this.state.comment} </span></div>
                <button >Comment</button>
            </div>
        )
    }
}