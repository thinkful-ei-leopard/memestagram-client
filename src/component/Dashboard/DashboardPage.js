import React, { Component } from 'react'

export default class DashboardPage extends Component {
    render() {
        return (
            <div>
                <div className='profImg'>Image</div><span>Username</span>
                <div className='MemePost'>Meme post</div>
                <span>❤️like: 420</span><span>Comment: 6 </span>
            </div>
        )
    }
}
