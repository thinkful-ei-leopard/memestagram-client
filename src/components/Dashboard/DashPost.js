import React, { Component } from 'react'
import {Link} from 'react-router-dom';

export default class DashboardPage extends Component {
    state={
        count:0,
        comment:0,
        like:'🤍'
    }

    addLike(){
    
    if(this.state.like==='🤍'){
     this.setState({
         like:'❤️',
         count:this.state.count+1
     })
        
    }else if(this.state.like==='❤️'){
        this.setState({
            like:'🤍',
            count:this.state.count-1
        })
        
    }
    
   }
       
    
    render() {
        
        return (
            <div>
                <div className='profImg'>Image</div><Link to='/photo'>Username</Link>
                <div className='MemePost'>Mmem post</div>
                 <div ><span role='img' aria-label='heart' onClick={() =>this.addLike()}>{this.state.like}</span>like: {this.state.count}<span> Comment: {this.state.comment} </span></div>
                <button >Comment</button>
            </div>
        )
    }
}