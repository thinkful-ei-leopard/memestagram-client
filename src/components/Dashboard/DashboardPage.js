import React, { Component } from 'react'
import DashPost from './DashPost'
import NavBar from '../NavBar/NavBar'
import MemeContext from '../../context/MemeContext'
import PostsService from '../../services/posts-service'


import './Dashboard.css'

export default class DashboardPage extends Component {
    state = {
       posts: [],
       error: null
    }

    static contextType = MemeContext

    componentDidMount(){
        this.context.clearError()
        PostsService.getPosts()
            .then(res => this.setState({posts: res}))
            .catch(this.context.setError)
    } 

    renderPosts() {
        const { posts = [] } = this.state

        return posts.map(post => 
            <DashPost
            key={post.id}
            post={post}
            />
        )
    }

    render() {
        const { error } = this.state
        return (
            <div>
                <NavBar />
                {error
                    ? <p>There was an error, please try again</p>
                    : this.renderPosts()
                }
            </div>
        )
    }
}
