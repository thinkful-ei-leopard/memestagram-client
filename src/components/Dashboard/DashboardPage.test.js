import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DashboardPage from './DashboardPage';
import DashPost from './DashPost';
import { BrowserRouter } from 'react-router-dom';

describe('Dashboard Page Component', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <BrowserRouter>
        <DashboardPage />
      </BrowserRouter>,
      div
    );
    ReactDom.unmountComponentAtNode(div);
  });

  it('renders Dashboard and post', () => {
    const post = {
      id: 1,
      userImg: 'https://i.insider.com/5c59e77ceb3ce80d46564023?width=600&format=jpeg&auto=webp',
      user_id: 1,
      username: 'Shrek',
      memeImg: 'https://live.staticflickr.com/5611/15033991623_66352974b6_z.jpg',
      description: 'test'
    } 

    const wrapper = shallow(
      <BrowserRouter>
        <DashboardPage>
          <DashPost
          key={post.id}
          post={post}
          />
        </DashboardPage>
      </BrowserRouter>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})