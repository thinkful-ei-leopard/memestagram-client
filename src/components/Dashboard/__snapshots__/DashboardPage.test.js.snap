// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page Component renders Dashboard and post 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <DashboardPage>
    <DashPost
      handlePhotoView={[Function]}
      handleUserView={[Function]}
      key="1"
      post={
        Object {
          "description": "test",
          "id": 1,
          "memeImg": "https://live.staticflickr.com/5611/15033991623_66352974b6_z.jpg",
          "userImg": "https://i.insider.com/5c59e77ceb3ce80d46564023?width=600&format=jpeg&auto=webp",
          "user_id": 1,
          "username": "Shrek",
        }
      }
    />
  </DashboardPage>
</Router>
`;
